Content.Server/Respawn/GeofencingSystem.cs

```diff
@@ -0,0 +1,164 @@
+using Content.Server.Administration.Logs;
+using Content.Server.Popups;
+using Content.Server.Respawn;
+using Content.Server.Shuttles.Systems;
+using Content.Shared.Database;
+using Content.Shared.Geofencing;
+using Robust.Shared.Timing;
+using Robust.Shared.Map;
+using System.Numerics;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Content.Server.Geofencing;
+
+public sealed class GeofencingSystem : EntitySystem
+{
+
+    [Dependency] private readonly IAdminLogManager _adminlogs = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly PopupSystem _popups = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly EmergencyShuttleSystem _emergencyShuttle = default!;
+    [Dependency] private readonly SpecialRespawnSystem _respawnSys = default!;
+
+
+    static private float _updateRate = 2; //How frequently to do our geofencing queries for perf reasons.
+    private TimeSpan _nextUpdate = TimeSpan.FromSeconds(_updateRate);
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<GeofencingComponent, MapInitEvent>(OnMapInit);
+
+    }
+
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        if (_nextUpdate > _timing.CurTime)
+            return;
+
+        _nextUpdate = _nextUpdate + TimeSpan.FromSeconds(_updateRate);
+
+        var fencequery = EntityQueryEnumerator<GeofencingComponent>();
```

Comment:
```suggestion
        var fenceQuery = EntityQueryEnumerator<GeofencingComponent>();
```

etc in the file, use camelCase consistently.